### PR TITLE
Fix #2504 App crashes when going to "Product Image" tab

### LINKED ISSUE #2504: App crashes when going to "Product Image" tab
- Scan this product: https://barcode.tec-it.com/en/EAN13?data=2115834795652
- Go to "Product Images" tab
- The app crashes

```
03-31 23:51:33.229 24541 24541 E AndroidRuntime: FATAL EXCEPTION: main
03-31 23:51:33.229 24541 24541 E AndroidRuntime: Process: openfoodfacts.github.scrachx.openfood, PID: 24541
03-31 23:51:33.229 24541 24541 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'int org.json.JSONArray.length()' on a null object reference
03-31 23:51:33.229 24541 24541 E AndroidRuntime: 	at openfoodfacts.github.scrachx.openfood.fragments.ProductPhotosFragment$1.onImageResponse(ProductPhotosFragment.java:93)
03-31 23:51:33.229 24541 24541 E AndroidRuntime: 	at openfoodfacts.github.scrachx.openfood.network.OpenFoodAPIClient$6.onResponse(OpenFoodAPIClient.java:344)
03-31 23:51:33.229 24541 24541 E AndroidRuntime: 	at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall$1$1.run(ExecutorCallAdapterFactory.java:70)
03-31 23:51:33.229 24541 24541 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:790)
03-31 23:51:33.229 24541 24541 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
03-31 23:51:33.229 24541 24541 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:164)
03-31 23:51:33.229 24541 24541 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6494)
03-31 23:51:33.229 24541 24541 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
03-31 23:51:33.229 24541 24541 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:440)
03-31 23:51:33.229 24541 24541 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
03-31 23:51:33.236   617  4993 W ActivityManager:   Force finishing activity openfoodfacts.github.scrachx.openfood/.views.ContinuousScanActivity
03-31 23:51:33.242   617   639 I ActivityManager: Showing crash dialog for package openfoodfacts.github.scrachx.openfood u0
```

### PR DESCRIPTION
**Pull request by deniger**
## Related issues and discussion
#fixes #2504 App crashes when going to "Product Image" tab

### CODE REVIEW

**Comment by teolemon** [comment]
@PrajwalM2212 is the original coder of this part

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/ProductPhotosFragment.java [hunk 1]
```diff
@@ -64,72 +55,56 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+                                    namesString.contains("o")) {
 
                                     continue;
-
                                 }
-                                imageNames.add(images.names().getString(i));
+                                imageNames.add(namesString);
                             } catch (JSONException e) {
-                                e.printStackTrace();
+                                Log.w(ProductPhotosFragment.class.getSimpleName(), "can get product / images in jon", e);
```
Comment (PrajwalM2212) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/ProductPhotosFragment.java]: `Log.w(ProductPhotosFragment.class.getSimpleName(), "can get product / images in json", e);`

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/ProductPhotosFragment.java [hunk 2]
```diff
@@ -64,72 +55,56 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
-                        e.printStackTrace();
-                    }
-
-                    if (images != null) {
+                // a json object referring to images
+                JSONObject images = null;
+                try {
+                    images = jsonObject.getJSONObject("product").getJSONObject("images");
+                } catch (JSONException e) {
+                    Log.w(ProductPhotosFragment.class.getSimpleName(), "can get product / images in jon", e);
```
Comment (PrajwalM2212) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/ProductPhotosFragment.java]: `Log.w(ProductPhotosFragment.class.getSimpleName(), "can get product / images in jon", e);`

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/ProductPhotosFragment.java [hunk 3]
```diff
@@ -64,72 +55,56 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
+                // a json object referring to images
+                JSONObject images = null;
+                try {
+                    images = jsonObject.getJSONObject("product").getJSONObject("images");
+                } catch (JSONException e) {
+                    Log.w(ProductPhotosFragment.class.getSimpleName(), "can't get product / images in json", e);
+                }
+                if (images != null) {
+                    final JSONArray names = images.names();
+                    if (names != null) {
```
Comment (PrajwalM2212) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/ProductPhotosFragment.java]: This is the heart of the change. Also nice refactoring work.

**Comment by teolemon** [comment]
Thanks for the review @PrajwalM2212 
Thanks for the refacto @deniger

**Pull request merged**